content/en/docs/features/control-plane/entity-tag.md

```diff
@@ -0,0 +1,217 @@
+
+Now, the first client, unaware of the resource change by the other client, makes a request to the resource using the cached ETAG value of 1000 from the previous run. The device responds with a code of `VALID`, indicating that the ETAG is still valid. However, the first client is unaware that the resource has been changed by another client, and the device's resource actually has same the ETAG.
+
+{{< /note >}}
+
+```c
+static uint64_t g_etag;
+
+static uint64_t etag_random() {
+    return (oc_random_value()%1000)+1;
+}
+
+void oc_etag_init()
+{
+    g_etag = oc_clock_time() + etag_random();
+}
+
+// Get new ETAG for changed resource
+uint64_t oc_etag_get(void)
+{
+    uint64_t now = oc_clock_time();
+    if (now > g_etag)
+    {
+        g_etag = now;
+    }
+    g_etag = g_etag + etag_random();
+    return g_etag;
+}
+
+// Load ETAGs from storage and clear storage
+bool oc_etag_load_and_clear()
+{
+    uint64_t loaded_etags[OC_MAX_APP_RESOURCES];
+    bool ok = oc_etags_load(&loaded_etags);
+    oc_etag_clear_storage();
+    if (!ok)
+    {
+        return false;
+    }
+    uint64_t etag = oc_clock_time();
+    for (int i = 0; i < num_loaded_etag; ++i)
+    {
+        if (etag < loaded_etags[i])
+        {
+            etag = loaded_etags[i];
+        }
+    }
+    // Reinitialize global ETAG value if it can be less than loaded ETAGs
+    g_etag = etag + etag_random();
+
+    // Set ETAG for all resources
+    for (/*resources in device*/)
+    {
+        if (/*ETAG is in loaded_etags*/)
+        {
+            oc_resource_set_etag(resource, etag);
+        }
+        else
+        {
+            oc_resource_set_etag(resource, oc_etag_get());
+        }
+    }
+    return true;
+}
+```
+
+### Updating ETAG on the Device
+
+There are two methods for updating the ETAG when a resource changes:
+
+- `oc_notify_observers`: Notifies all observers that the resource has changed, what internally invokes `oc_resource_update_etag`.
+- `oc_resource_update_etag`: Updates the ETAG without notifying observers about the content change. This could be useful for non-observable resources.
+
+### ETAG Usage by Clients
+
+To optimize network traffic, clients can cache device resources as a proxy. This caching process requires the client to store the cached ETAG received with GET responses. To determine if the ETAG is still valid, the client can send the ETAG in a request to the device. If the ETAG is valid, the device will respond with a `VALID` code and no body. If the ETAG is not valid, the device will respond with a `CONTENT` code containing the current ETAG and the corresponding body.
```

Comment:
To optimize network traffic, clients can cache device resources as a proxy. This caching process requires the client to store the cached ETAG received with GET responses. 

-> To optimize network traffic, clients can cache received ETAGs from GET responses.

Reply:
fixed